src/main/java/spoon/reflect/declaration/CtClass.java

```diff
@@ -49,6 +49,18 @@
 	@PropertyGetter(role = CONSTRUCTOR)
 	CtConstructor<T> getConstructor(CtTypeReference<?>... parameterTypes);
 
+	/**
+	 * Returns the constructor of the class that takes the given signature.
+	 * e.g. java.util.HashSet has constructor with no parameters `()`
+	 * e.g. java.util.HashSet has constructor with a int and a float parameters `(int,float)`
+	 * e.g. java.util.HashSet has constructor with a java.util.Collection parameter `(java.util.Collection)`
+	 *
+	 * Derived from {@link #getTypeMembers()}
+	 */
+	@DerivedProperty
+	@PropertyGetter(role = CONSTRUCTOR)
+	CtConstructor<T> getConstructorBySignature(String signature);
```

Comment:
I don't quite see how this relates to the title and scope of this PR. This should be done in a separate PR IMHO.

Reply:
As a CtConstructor has a CtPath ends with `#constructor[signature=(int)]`, I need this function to get the CtElement.